File: 최민정/문제/1주차/10845/10845.py
import sys
input = sys.stdin.readline

N = int(input())
queue=[]

for i in range(N):
  tmp=input().split()

  if tmp[0]=='push':
    queue.append(tmp[1])
  elif tmp[0]=='pop':
    if queue:
      print(queue.pop(0))
    else:
      print(-1)
  elif tmp[0]=='size':
    print(len(queue))
  elif tmp[0]=='empty':
    if queue:
      print(0)
    else:
      print(1)
  elif tmp[0]=='front':
    if queue:
      print(queue[0])
    else:
      print(-1)
  elif tmp[0]=='back':
    if queue:
      print(queue[-1])
    else:
      print(-1)

File: 이지홍/문제/1주차/10845/10845.js
const fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const iter = Number(input.shift());

// 큐 생성
let queue = [];

let ans = [];

for(let i = 0; i < iter; i++){
    let [cmd, num] = input[i].split(" ");
    
    if(cmd === "push"){
        queue.push(num);
    } else if(cmd === "pop"){
        if(queue.length > 0){
            ans.push(queue.shift());
        } else {
            ans.push(-1);
        }
    } else if(cmd === "size"){
        ans.push(queue.length);
    } else if(cmd === "empty"){
        if(queue.length > 0){
            ans.push(0);
        } else {
            ans.push(1);
        }
    } else if(cmd === "front"){
        if(queue.length > 0){
            ans.push(queue[0]);
        } else {
            ans.push(-1);
        }
    } else{
        if(queue.length > 0){
            ans.push(queue[queue.length - 1]);
        } else{
            ans.push(-1);
        }
    }
}

console.log(ans.join("\n"));